include/range/v3/numeric/adjacent_difference.hpp

```diff
@@ -81,6 +55,7 @@ namespace ranges
             operator()(I begin, S end, O result, S2 end_result, BOp bop = BOp{},
                        P proj = P{}) const
             {
+                // BUGBUG think about the use of coerce here.
                 using V = value_type_t<I>;
                 using X = concepts::Invocable::result_t<P, V>;
```

Comment:
Pretty sure this should be `concepts::Invocable::result_t<P&, V>` here. Mind changing that?